### ISSUE TITLE
Accessibility :: Dates are not reachable using speak over	

### ISSUE THREAD

**Issue by ben-caplan** [Hacker0x01/react-datepicker#2003]
**Describe the bug**
Per the WCAG A11y specification,  a user should be able to hear the dates as he/she navigates through the Date Picker.

**To Reproduce**

Use the date picker with any screen reader
- Chrome/JAWS
- Firefox/NVDA
- Safari/VoiceOver
- IE/JAWS

**Expected behavior**
User is able to hear the dates as they navigate through the Date Picker

**Desktop (please complete the following information):**

- OS: All
- Browser: All
- Version: All (but we are using 2.10.1

**Additional context**
We would like to contribute to your project by addressing this A11y issue. Please let us know asap if this update is something you would be open to us working on.

**Comment by martijnrusschen** [comment]
Letting this component become WCAG A11y compliant would be a huge improvement. Any work that contributes to that would be welcomed of course :) Feel free to submit a PR and I'll make sure to check it out and provide help getting it released.

**Comment by ben-caplan** [comment]
Wonderful, thanks! We will get a PR set up ASAP. Are there any code standards or practices that we should be mindful of while making our enhancements?

**Comment by martijnrusschen** [comment]
There are a few linters (stuff like ESlint) but those are enforced via the CI, but with any modern editor, those should be picked up automatically. Nothing special at least

**Comment by ben-caplan** [comment]
Sounds good, thanks!

**Comment by krv5vignesh** [comment]
Issue still exists - Not able to foucs all the dates when screen reader is open. Using the Up/down arrow key while the screen reader (NVDA) enabled, fouces only few dates on the calendar. 

**Expected behavior:** 
Using Up/down arrow key while using the screen reader should focus all the dates.

**Current Behavior**

Only few dates are focused at random and the skipped dates are read by the screen reader all together with the focused date.

Example: https://imgur.com/a/j0avtaB

@ben-caplan @dan-diaz  Looks this issue was not fixed my your PR. Can you confirm?
